ceph/basic-pynfs-cthon.sh

```diff
@@ -0,0 +1,279 @@
+#!/bin/sh
+
+# if any command fails, the script should exit
+set -e
+
+# enable some more output
+set -x
+
+# these variables need to be set
+[ -n "${GERRIT_HOST}" ]
+[ -n "${GERRIT_PROJECT}" ]
+[ -n "${GERRIT_REFSPEC}" ]
+
+# only use https for now
+GIT_REPO="https://${GERRIT_HOST}/${GERRIT_PROJECT}"
+
+# enable the Storage SIG Gluster and Ceph repositories
+dnf -y install centos-release-ceph epel-release
+
+BUILDREQUIRES="git bison cmake dbus-devel flex gcc-c++ krb5-devel libacl-devel libblkid-devel libcap-devel redhat-rpm-config rpm-build xfsprogs-devel lvm2"
+
+BUILDREQUIRES_EXTRA="libnsl2-devel libnfsidmap-devel libwbclient-devel userspace-rcu-devel"
+
+# basic packages to install
+case "${CENTOS_VERSION}" in
+    7)
+        yum install -y ${BUILDREQUIRES} ${BUILDREQUIRES_EXTRA} python2-devel
+    ;;
+    8s)
+        yum install -y ${BUILDREQUIRES}
+        yum install --enablerepo=powertools -y ${BUILDREQUIRES_EXTRA}
+        yum install -y libcephfs-devel
+    ;;
+    9s)
+       yum install -y ${BUILDREQUIRES}
+       yum install --enablerepo=crb -y ${BUILDREQUIRES_EXTRA}
+       yum install -y libcephfs-devel
+    ;;
+esac
+
+git clone --depth=1 ${GIT_REPO}
+cd $(basename "${GERRIT_PROJECT}")
+git fetch origin ${GERRIT_REFSPEC} && git checkout FETCH_HEAD
+
+# update libntirpc
+git submodule update --recursive --init || git submodule sync
+
+# cleanup old build dir
+[ -d build ] && rm -rf build
+
+mkdir build
+cd build
+
+( cmake ../src -DCMAKE_BUILD_TYPE=Maintainer -DUSE_FSAL_GLUSTER=OFF -DUSE_FSAL_CEPH=ON -DUSE_FSAL_RGW=OFF -DUSE_DBUS=ON -DUSE_ADMIN_TOOLS=ON && make) || touch FAILED
+make install
+
+# dont vote if the subject of the last change includes the word "WIP"
+if ( git log --oneline -1 | grep -q -i -w 'WIP' )
+then
+    echo "Change marked as WIP, not posting result to GerritHub."
+    touch WIP
+fi
+
+# If failure found during build, return the status and skip proceeding
+# to ceph configuration
+
+
+# we accept different return values
+# 0 - SUCCESS + VOTE
+# 1 - FAILED + VOTE
+# 10 - SUCCESS + REPORT ONLY (NO VOTE)
+# 11 - FAILED + REPORT ONLY (NO VOTE)
+RET=0
+if [ -e FAILED ]
+then
+	exit ${RET}
+fi
+if [ -e WIP ]
+then
+	RET=$[RET + 10]
+	exit ${RET}
+fi
+
+# Create a virtual disk file (for OSD storage):
+truncate -s 35G /tmp/ceph-disk.img
+losetup -f /tmp/ceph-disk.img  # Attaches as a loop device (e.g., /dev/loop0)
+
+pvcreate /dev/loop0
+vgcreate ceph-vg /dev/loop0
+lvcreate -L 10G -n osd1 ceph-vg
+lvcreate -L 10G -n osd2 ceph-vg
+lvcreate -L 10G -n osd3 ceph-vg
+
+# Install and configure ceph cluster
+dnf install -y cephadm
+cephadm add-repo --release squid
+dnf install -y ceph
+cephadm bootstrap --mon-ip $(hostname -I | awk '{print $1}') --single-host-defaults --allow-fqdn-hostname
```

Comment:
Let's also include `skip-monitoring-stack` as our focus is on cephFS + NFS Ganesha.